worksheet_D/oxo.py

```diff
@@ -0,0 +1,109 @@
+class OxoBoard:
+    def __init__(self):
+        # Initialises the board array
+        self.boardgrid = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+
+    def get_square(self, x, y):
+        # Returns the contents of the specified square
+        return self.boardgrid[x][y]
+
+    def set_square(self, x, y, mark):
+        # Fills empty squares with mark and returns True, if square is not empty returns False
+        if self.boardgrid[x][y] == 0:
+            self.boardgrid[x][y] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        # Checks every square to see if it is empty and returns True if none of them are empty.
+        for x in xrange(3):
+            for y in xrange(3):
+                if self.boardgrid[x][y] == 0:
+                    return False
+        return True
+
+    def get_winner(self):
+        # Checks if 3 squares in a row, column or diagonal are the same but not empty and returns the value in those squares (1 or 2).
+        for i in xrange(3):
+            # Horizontal win
+            if self.boardgrid[0][i] == self.boardgrid[1][i] == self.boardgrid[2][i]:
+                if self.boardgrid[2][i] != 0:       # Checks if squares aren't empty
+                    return self.boardgrid[2][i]
+            # Vertical win
+            elif self.boardgrid[i][0] == self.boardgrid[i][1] == self.boardgrid[i][2]:
+                if self.boardgrid[i][2] != 0:
+                    return self.boardgrid[i][2]
+            #Diagonal win
+            elif self.boardgrid[0][0] == self.boardgrid[1][1] == self.boardgrid[2][2]:

```

Comment:
The diagonal checks do not need to be inside the `for` loop -- means you are checking the diagonals three times